### PR TITLE
Frontend conv pool support

### PR DESCRIPTION
**Pull request by fumihwh**

### CODE REVIEW

File: onnx_tf/frontend.py [hunk 1]
```diff
@@ -189,6 +213,46 @@ def _bin_op(cls, node, onnx_op):
...
+  @classmethod
+  def handle_avg_pool(cls, node, consts):
+    return cls._pool_op(node, "AveragePool")
+
+  @classmethod
+  def handle_conv2_d(cls, node, consts):
+    auto_pad = node.attr["padding"].decode("UTF-8")
+    auto_pad = "SAME_UPPER" if auto_pad == "SAME" else auto_pad
+    data_format = node.attr["data_format"].decode("UTF-8")
+    nc_index = list(map(lambda i: data_format.find(i), "NC"))
```
Comment (tjingrant) [inline onnx_tf/frontend.py]: The logic here is a bit error-prone (e.g., you shouldn't use i to indicate a character; it's quite difficult for others to read). It seems like nc_index is calculated just to mask off indicies in spatial_index. I would just write:
```
spatial_indices = [i for i in range(len(data_format))
				   if data_format[i] not in ["N", "C"]]
```

Comment (fumihwh) [reply onnx_tf/frontend.py]: Great...
Why did I get nc_index to get spatial_index.....weird..

**Comment by tjingrant** [comment]
Do you need help with this one?

**Comment by fumihwh** [comment]
Today I will work on this.

**Comment by fumihwh** [comment]
@tjingrant I am facing a little problem.
In tf, there are `conv1d`, `conv2d`, `conv3d` which in onnx is just `conv`.
How do I make mapping for them in frontend?
In current mechanism, I can change `ONNX_TO_HANDLER` to key: list.
What do you think?

https://github.com/onnx/onnx-tensorflow/pull/67/files#diff-db225cdb058d0ce5d91e7e0434e29631R18
```
  ONNX_TO_HANDLER = {
    "conv": ["conv1_d", "conv2_d", "conv3_d"],}
```

**Comment by tjingrant** [comment]
Yes, I think that's fair. Since we are forced into a many-to-one conversion.

**Comment by tjingrant** [comment]
I thought ONNX community has stepped back from enforcing consumed tensor:
```
ERROR: test_batch_normalization (test_node.TestNode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/onnx/onnx-tensorflow/test/backend/test_node.py", line 141, in test_batch_normalization
    output = run_node(node_def, [x, scale, bias, m, v])
  File "/home/travis/build/onnx/onnx-tensorflow/onnx_tf/backend.py", line 223, in run_node
    super(TensorflowBackendBase, cls).run_node(node, inputs, device)
  File "/home/travis/work/onnx/onnx/onnx/backend/base.py", line 70, in run_node
    onnx.checker.check_node(node)
  File "/home/travis/work/onnx/onnx/onnx/checker.py", line 45, in checker
    proto.SerializeToString(), ctx)
onnx.onnx_cpp2py_export.checker.ValidationError: Input index 3 is set to consumed but is not supported by op BatchNormalization
```

**Comment by fumihwh** [comment]
Oh..no....
It's a new opset version 6....
Let me fix it.

**Comment by fumihwh** [comment]
Finally, I think this pr could be landed.

**Comment by tjingrant** [comment]
Would you like to merge this one?

**Comment by fumihwh** [comment]
Yes. Support conv and pool will make frontend more practical. 
If everything goods, I want to merge this.

**Comment by tjingrant** [comment]
(I was just wondering why you haven't done so.)

**Comment by fumihwh** [comment]
I made lots of changes after your last review. So maybe there is any comment you want to add.......

**Comment by tjingrant** [comment]
@fumihwh I think I'm up to date on this one; feel free to merge it any time.

**Pull request merged**